tests/functional/invalid_configs/test_invalid_where.py

```diff
@@ -0,0 +1,102 @@
+from struct import pack
+import os
+import pytest
+from dbt.tests.util import run_dbt
+
+# our file contents
+from tests.functional.fixtures import (
+    fact_orders_source_csv,
+    fact_orders_sql,
+    fact_orders_yml,
+)
+
+# models/invalid_where.sql
+invalid_where_sql = """
+select *
+from 
+{{ metrics.calculate(metric('invalid_where'), 
+    grain='month',
+    dimensions=['had_discount'],
+    where=["order_country='Japan'"]
+    )
+}}
+"""
+
+# models/invalid_where.yml
+invalid_where_yml = """
+version: 2 
+models:
+  - name: invalid_where
+    tests: 
+      - dbt_utils.equality:
+          compare_model: ref('invalid_where__expected')
+metrics:
+  - name: invalid_where
+    model: ref('fact_orders')
+    label: Total Discount ($)
+    timestamp: order_date
+    time_grains: [day, week, month]
+    type: sum
+    sql: order_total
+    dimensions:
+      - had_discount
+      - order_country
+"""
+
+class TestInvalidWhereMetric:
+
+    # configuration in dbt_project.yml
+    if os.getenv('dbt_target') == 'bigquery':
+        @pytest.fixture(scope="class")
+        def project_config_update(self):
+            return {
+            "name": "example",
+            "models": {"+materialized": "table"}
+            }
+    else: 
+        @pytest.fixture(scope="class")
+        def project_config_update(self):
+            return {
+            "name": "example",
+            "models": {"+materialized": "view"}
+            }  
```

Comment:
Does this mean that BQ needs a table for some reason and the others don't care? If so, any reason to not just make everything a table? Less branches of logic == less headaches IMO

Reply:
So fun discovery during this process! BigQuery apparently has a behind the scenes `complexity rating` for every query. And the rating for views carries over to all queries that use it. We were getting resource constraint errors on BigQuery that I resolved by turning them into tables - this lowered performance for Bigquery but allowed it to run.

We don't do this on other DWHs because the performance takes an even bigger hit due to Pytest being so slow.

Reply:
okey dokey! A comment to that effect might not go astray. 

other than that, all i can say is 

![bigqueryyyy](https://user-images.githubusercontent.com/7335046/182222189-f8d55054-1451-4e7a-a61f-6dd62fef3e15.gif)